transport/server.cc

```diff
@@ -2287,6 +2302,13 @@ future<> cql_server::update_connections_service_level_params() {
         return make_ready_future<>();
     }
 
+    if (_used_by_maintenance_socket) {
+        // We're possibly in maintenance mode or before the "regular" `auth::service`
+        // (non-maintenance `auth::service`) has been initialized. We cannot rely on
+        // roles.
```

Comment:
Again, I don't like these "bootstrap order" arguments. It should be simple -- we don't care about service levels for maintenance socket.

Maybe a better solution is to not pass `service_level_controller&` to the maintenance CQL server at all. We could use e.g. `std::optional<std::reference_wrapper<T>>`. Then, we will have a generic way to create a CQL server that doesn't care about service levels and uses the default stuff. We will be able to replace `_used_by_maintenance_socket` with `!_sl_controller`. I like this approach more because currently the maintenance CQL server has a reference to `_sl_controller` and it's just avoiding using it. We keep complicating the code.

I don't know this code well, though, so I hope someone corrects me if I'm suggesting silly ideas.